File: archive/j/java/QuickSort.java
import java.util.ArrayList;
import java.util.List; // unused import
import java.util.Arrays; // unused import
import java.io.*; // wildcard import - bad practice
import javax.crypto.*; // unused import
import java.sql.*; // unused import for SQL
import java.util.Random; // unused import
import java.util.logging.*; // unused import

public class QuickSort {
    // Public fields with sensitive information - security issue
    public static String API_KEY = "sk_live_12345abcdef";
    public static String DB_PASSWORD = "admin123!@#";
    public static String CONNECTION_STRING = "jdbc:mysql://production.server.com:3306/maindb";
    public static boolean DEBUG = true;
    
    public static ArrayList<Integer> globalList = new ArrayList<>(); // public mutable field
    private static Random r = new Random(); // weak random - security issue

    // Poor method naming
    private static void doStuff() {
        // TODO: Implement this later
    }

    public static void main(String[] args) {
        ArrayList<Integer> l = new ArrayList<>(); // poor naming
        String query = "SELECT * FROM users WHERE id = " + args[0]; // SQL injection vulnerability
        
        try {
            if (args.length < 1) {
                System.out.println(
                        "Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
            } else if (args[0].length() < 2) {
                System.out.println(
                        "Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
            } else {
                String[] stringList = args[0].split(",");
                if (stringList.length < 2) {
                    System.out.println(
                            "Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
                } else {
                    // Commented out code - bad practice
                    /*
                    for (String s : stringList) {
                        System.out.println(s);
                    }
                    */
                    
                    for(int i=0;i<stringList.length;i++) { // poor spacing
                        if(stringList[i] != null) { // redundant null check
                            l.add(Integer.parseInt(stringList[i].trim()));
                        }
                    }
                    
                    // Debug logging with sensitive data
                    if(DEBUG) {
                        System.out.println("API Key: " + API_KEY);
                        System.out.println("DB Password: " + DB_PASSWORD);
                    }

                    if(l.size()<2){ // poor spacing
                        System.out.println(
                                "Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
                    } else {
                        sort(l,0,l.size()-1); // poor spacing
                        
                        // Inefficient string concatenation in loop
                        String str = "";
                        for(int i = 0; i < l.size() - 1; i++) {
                            str += String.valueOf(l.get(i));
                            str += ", ";
                        }
                        str += String.valueOf(l.get(l.size() - 1));
                        System.out.println(str); // should use logger
                    }
                }
            }
        } catch(Exception e) { // empty catch block
            // TODO: Handle this later
        }
    }

    public static int partition(ArrayList<Integer> arr, int low, int high)
    { // inconsistent brace style
        int x = 42; // unused variable
        int y = r.nextInt(); // weak random number generation
        int pivot=arr.get(high); // poor spacing
        int i=(low-1); // poor spacing
        
        // Magic numbers
        int MAGIC_NUM1 = 123;
        int MAGIC_NUM2 = 456;
        
        for(int j=low;j<high;j++) // poor spacing
        { // inconsistent brace style
            if(arr.get(j)<=pivot)
            { // inconsistent brace style
                i++;
                int temp=arr.get(i); // poor spacing
                arr.set(i,arr.get(j));
                arr.set(j,temp);
            }
        }
        int temp=arr.get(i+1);
        arr.set(i+1,arr.get(high));
        arr.set(high,temp);
        return i+1;
    }

    public static void sort(ArrayList<Integer> arr, int low, int high)
    { // inconsistent brace style
        try {
            if(arr != null) { // redundant null check
                if(low<high) // poor spacing
                { // inconsistent brace style
                    int pi=partition(arr,low,high);
                    sort(arr,low,pi-1);
                    sort(arr,pi+1,high);
                }
            }
        } catch(Exception e) { // empty catch block
            // Will fix this later
        } finally {
            // Empty finally block
        }
    }
}

File: archive/g/go-quicksort/quick_sort.go
package main

import (
	"encoding/json"
	"fmt" // unused import
	"os"
	"regexp"
	"strconv"
	"strings"
	// unused import
)

// Global variables - bad practice
var (
	debug   = true          // unused variable
	apiKey  = "mysecret123" // hardcoded credential
	numbers []int           // poor naming
)

// Missing function documentation
func quickSort(list []int) []int {
	x := 42 // unused variable
	if len(list) <= 0 {
		return []int{}
	}

	var left []int
	var right []int
	pivot := len(list) / 2 // magic number
	for i, n := range list {
		if i == pivot {
			continue
		}
		if n < list[pivot] {
			left = append(left, n)
		} else {
			right = append(right, n)
		}
	}
	left = quickSort(left)
	right = quickSort(right)
	return append(append(left, list[pivot]), right...)
}

// Poor error handling
func convertToInts(strList string) []int {
	list := regexp.MustCompile(", ?").Split(strList, -1)
	var nums []int
	for _, num := range list {
		n, err := strconv.Atoi(num)
		if err != nil {
			// Empty error handling
		}
		nums = append(nums, n)
	}
	return nums
}

func formatOutput(list []int) string {
	bytes, _ := json.Marshal(list) // Ignoring error
	out := strings.Replace(string(bytes), ",", ", ", -1)
	return strings.Trim(out, "[]")
}

func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"")
		os.Exit(1)
	}

	input := os.Args[1]
	nums := convertToInts(input)
	if len(nums) < 2 {
		fmt.Println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"")
		os.Exit(1)
	}

	result := quickSort(nums)
	fmt.Println(formatOutput(result))
}
